src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -906,6 +897,113 @@ static bool usesReceiver(BoundExpression argument)
             }
         }
 
+        private RefKind GetRefKindForCapturedReceiverTemp(BoundExpression rewrittenReceiver)
+        {
+            Debug.Assert(rewrittenReceiver.Type is not null);
+
+            RefKind refKind = rewrittenReceiver.GetRefKind();
+            if (refKind == RefKind.RefReadOnlyParameter)
+            {
+                refKind = RefKind.Ref;
```

Comment:
>Ref [](http://example.com/codeflow?start=34&length=3)

Dropping `readonly` aspect of the reference looks suspicious. We probably should use `RefReadOnly` instead. I think it should be possible to come up with a scenario for which the difference is observable.